datafusion/core/tests/sql/expr.rs

```diff
@@ -491,6 +491,16 @@ async fn test_crypto_expressions() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_array_index() -> Result<()> {
+    test_expression!("([5,4,3,2,1])[1]", "5");
+    test_expression!("([5,4,3,2,1])[5]", "1");
+    test_expression!("([5,4,3,2,1])[100]", "NULL");
+    test_expression!("([5,4,3,2,1])[-1]", "NULL");
```

Comment:
I wonder if you want to potentially try nested lists. Something like

```suggestion
    test_expression!("([5,4,3,2,1])[-1]", "NULL");
    test_expression!("([[123],[4,5,6]])[2]", "[4,5,6]");
```

Reply:
Right now, it's not possible to define multi-dimension arrays via SQL.

Reply:
Oh, Nope, I was wrong.